### DISCUSSION TITLE
"could not receive: [6] Not Leader For Partition: the client attempted to send messages to a replica that is not the leader for some partition, the client's metadata are likely out of date:

### DISCUSSION THREAD

**Discussion by anil9030** [strimzi/strimzi-kafka-operator#6454] [General]
one of our micro services (Metadata Change Detection) has been not able to create topics until we restarted that pod.
any idea why this error showed in logs

**Comment by scholzj** [comment]
AFAIK, That suggests that the client had stale metadata about the Kafka cluster.

**Reply by anil9030** [reply]
what would be reason for that? it never happened before

**Reply by scholzj** [reply]
I don't know -> the client is normally responsible for handling the metadata automatically. I guess it can be some bug or glitch?

**Reply by scholzj** [reply]
Thi leadership can change in the Kafka cluster for various reasons. The client should normally detect it (from errors like this), update the metadata and continue. So I'm not sure seeing this error itself is an issue. But it should normally recover from this automatically without restarting the pods.

**Reply by anil9030** [reply]
can you review this error in external kafka setup https://github.com/RedHatInsights/edge-api/pull/469
seems like they found some fix for it but can that be implemented here in strimzi

**Reply by scholzj** [reply]
Sorry, I'm not sure I see the relationship.

**Reply by anil9030** [reply]
shall i raise an issue with this? here?

**Reply by scholzj** [reply]
As I said, I think this is an issue of the client, not Strimzi.
